### PR TITLE
adds client method for getting 'similar' related videos via embedCode

### PR DESCRIPTION
**Pull request by aljachimiak**
Adds a Client method to retrieve videos that are similar to a provided embedCode.

Uses the Ooyala backlot [discovery api](http://help.ooyala.com/video-platform/concepts/content_discovery_summary_of_routes.html).

### CODE REVIEW

**Review by erkattak — COMMENTED** [review]
Seems good to me so far. I see you added example responses, but I don't see the tests for those responses.

**Comment by aljachimiak** [comment]
Hmm.... It seems that tests would be more apt once there was a method for handling the client response and transforming the data into oddworks objects.

At this point we haven't nailed down a related strategy, so I hesitated at this step.

The next logical step seems to be turning the results of the related calls into collections.

The only question that remains would be "Use three separate handlers for each of the related types or use One related handler that results in an appropriate collection?"

File: lib/fetch-popular-collection.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+		// First, get the discovery object from Ooyala.
+		return client.getPopularRelated(creds)
+			.then(popular => {
+				if (popular) {
+					// If the popular results exist, cast it to an Oddworks collection.
+
+					// There should only be one popular collection
+					spec.id = 'spec-ooyala-discovery-popular';
+					collection = transform(spec, {name: 'Popular Videos'});
```
Comment (blainsmith) [inline lib/fetch-popular-collection.js]: Thoughts on making `'Popular Videos'` configurable from the spec so this isn't hardcoded?

File: lib/fetch-trending-collection.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+		// First, get the discovery object from Ooyala.
+		return client.getTrendingRelated(creds)
+			.then(trending => {
+				if (trending) {
+					// If the trending results exist, cast it to an Oddworks collection.
+
+					// There should only be one trending collection
+					spec.id = 'spec-ooyala-discovery-trending';
+					collection = transform(spec, {name: 'Trending Videos'});
```
Comment (blainsmith) [inline lib/fetch-trending-collection.js]: Same idea here for `'Trending Videos'`

**Review by blainsmith — COMMENTED** [review]
Moving those 2 strings into the spec would let those be more configurable and the spec is the perfect place for those.

**Comment by kixxauth** [comment]
The example responses I started with in here were just to use as a reference for development. I never intended to use them in tests fwiw.

File: lib/fetch-popular-collection.js (hunk 1 above)
Comment (kixxauth) [reply lib/fetch-popular-collection.js]: Yeah, and maybe the safe default could be 'Popular Videos'

File: lib/fetch-trending-collection.js (hunk 2 above)
Comment (kixxauth) [reply lib/fetch-trending-collection.js]: Yes, also with a safe default.

**Pull request merged**